### ISSUE TITLE
Issue parsing some "Part X of X" filenames

### ISSUE THREAD

**Issue by Colin1224** [Casvt/Kapowarr#194]
**Description of the bug**

There seems to be an issue parsing filenames that contain issue names with "Part X of X" in them. I was getting [Batgirls](https://comicvine.gamespot.com/batgirls/4050-140426/), and I was able to import all issues, but for some reason issues 7 and 8 were improperly being imported as issues 1 and 2.

**Screenshots**

![image](https://github.com/user-attachments/assets/c94491b8-5e32-4ede-9020-33c1fa6d087c)

**To Reproduce**

1. Set file naming to `{clean_series_name} - #{issue_number} - {issue_title}`
2. Add Volume https://comicvine.gamespot.com/batgirls/4050-140426/
3. Download issue 7

**Expected behaviour**

I tried manually removing the issue titles from issues 7 and 8 and rescanned, and they were properly imported as issues 7 and 8. I then used the rename feature to rename them back and then rescanned and Kapowarr falsely interpreted them as issues 1 and 2 again. I then tried removing just the "of 2" part of the issue title for both, and they imported as the correct issues, so there seems to be some issue around how the filename parser worked involving multipart issues. Though interestingly issues 4, 5, and 6 also have the "of 6" part of their issues names, but there was no problem with those, so I don't know what the problem is. I was also able to reproduce a similar issue with [Batman](https://comicvine.gamespot.com/batman/4050-91273/) issue 112, but in that case it was properly downloaded and named but not imported into Kapowarr at all, not even as the incorrect issue like with Batgirls issue 7.

**Version info**

```
Kapowarr version 	v1.0.0

Python version 	3.8.17.final.0

Database version 	27

Database location 	/app/db/Kapowarr.db

Data folder 	/app
```

Hosted on Docker for windows, WSL backend. Accessed via firefox.

**Comment by Casvt** [comment]
Just to give you a little insight into what the file parsing algorithm sees (it doesn't matter that the exact data, like ID's or dates, aren't correct):

```python
# Batgirls/Volume 1 (2022)/Batgirls - #007 - One Way or Another, Part 1 of 2 (2022-02-01) [898184].cbz

{
    "series": "Batgirls #007 One Way or Another, Part",
    "year": 2022,
    "volume_number": 1,
    "special_version": null,
    "issue_number": 1.0,
    "annual": false
}

# Batgirls/Volume 1 (2022)/Batgirls - #007 - One Way or Another, Part 1 (2022-02-01) [898184].cbz

{
    "series": "Batgirls",
    "year": 2022,
    "volume_number": 1,
    "special_version": null,
    "issue_number": 7.0,
    "annual": false
}

# Batgirls/Volume 1 (2022)/Batgirls - #005 - One Way or Another, Part 5 of 6 (2022-02-01) [898184].cbz

{
    "series": "Batgirls #005 One Way or Another, Part",
    "year": 2022,
    "volume_number": 1,
    "special_version": null,
    "issue_number": 5.0,
    "annual": false
}
```

You can see that the algorithm thinks the part is the issue number, but only when it's `of N`. The result for the last example is completely wrong, but it still happens to match correctly because the year and issue number (which was taken from the part number, but the part number and the issue number happen to be the same) match and Kapowarr doesn't look at the series value. So those files match because of pure luck.

I fixed the algorithm in commit 97957d1106924f473ab49e29f0113388090f0478. See below:

```python
# Batgirls/Volume 1 (2022)/Batgirls - #007 - One Way or Another, Part 1 of 2 (2022-02-01) [898184].cbz

{
    "series": "Batgirls",
    "year": 2022,
    "volume_number": 1,
    "special_version": null,
    "issue_number": 7.0,
    "annual": false
}

# Batman - #112 - Fear State Part 1; DIY Part 1 of 3 (2022-02-01) [898184]

{
    "series": "Batman",
    "year": 2022,
    "volume_number": 1,
    "special_version": null,
    "issue_number": 112.0,
    "annual": false
}
```

**Comment by Colin1224** [comment]
Awesome, thanks for the quick fix.
